crates/aptos-dkg/src/range_proofs/dekart_univariate_v2.rs

```diff
@@ -0,0 +1,809 @@
+        }
+    }
+
+    #[allow(non_snake_case)]
+    fn verify(
+        &self,
+        vk: &Self::VerificationKey,
+        n: usize,
+        ell: usize,
+        comm: &Self::Commitment,
+        fs_t: &mut merlin::Transcript,
+    ) -> anyhow::Result<()> {
+        // TODO: Do we want vk to have a max_ell?? If so:
+        // assert!(
+        //     ell <= vk.max_ell,
+        //     "ell (got {}) must be ≤ max_ell (which is {})",
+        //     ell,
+        //     vk.max_ell
+        // );
+
+        // Step 1
+        let VerificationKey {
+            xi_1,
+            zeroth_lagr,
+            vk_hkzg,
+            roots_of_unity,
+            verifier_precomputed,
+        } = vk;
+        let Proof {
+            hatC,
+            pi_PoK,
+            Cj,
+            D,
+            a,
+            a_h,
+            a_js,
+            pi_gamma,
+        } = self;
+
+        // Step 2a
+        fiat_shamir::append_initial_data(fs_t, DST, vk, n, ell, &comm);
+
+        // Step 2b
+        fiat_shamir::append_hat_f_commitment::<E>(fs_t, &hatC);
+
+        // Step 3
+        two_term_msm::Homomorphism {
+            base_1: *zeroth_lagr,
+            base_2: *xi_1,
+        }
+        .verify(&(two_term_msm::CodomainShape(*hatC - comm.0)), pi_PoK, fs_t)?;
+
+        // Step 4a
+        fiat_shamir::append_sigma_proof(fs_t, &pi_PoK);
+
+        // Step 4b
+        fiat_shamir::append_f_j_commitments::<E>(fs_t, &Cj);
+
+        // Step 5
+        let (beta, beta_js) = fiat_shamir::get_beta_challenges::<E>(fs_t, ell);
+
+        // Step 6
+        fiat_shamir::append_h_commitment::<E>(fs_t, &D);
+
+        // Step 7
+        let (mu, mu_h, mu_js) = fiat_shamir::get_mu_challenges::<E>(fs_t, ell);
+
+        // Step 8
+        let U_bases: Vec<E::G1Affine> = {
+            let mut v = Vec::with_capacity(2 + Cj.len());
+            v.push(*hatC);
+            v.push(*D);
+            v.extend_from_slice(&Cj);
+            E::G1::normalize_batch(&v)
+        };
+
+        let U_scalars: Vec<E::ScalarField> = {
+            let mut v = Vec::with_capacity(2 + mu_js.len());
+            v.push(mu);
+            v.push(mu_h);
+            v.extend_from_slice(&mu_js);
+            v
+        };
+
+        let U = E::G1::msm(&U_bases, &U_scalars).expect("Failed to compute MSM in DeKARTv2");
+
+        // Step 9
+        let gamma = fiat_shamir::get_gamma_challenge::<E>(fs_t, &roots_of_unity);
+
+        // Step 10
+        let a_u = *a * mu
+            + *a_h * mu_h
+            + a_js
+                .iter()
+                .zip(&mu_js)
+                .map(|(&a_j, &mu_j)| a_j * mu_j)
+                .sum::<E::ScalarField>();
+
+        univariate_hiding_kzg::CommitmentHomomorphism::verify(
+            vk_hkzg.clone(),
+            univariate_hiding_kzg::Commitment(U),
+            gamma,
+            a_u,
+            pi_gamma.clone(),
+        )?;
+
+        // Step 11
+        let num_omegas = roots_of_unity.len();
+
```

Comment:
Here (and below) it would be helpful to add comments with LaTeX indicating which part of the LHS/RHS you are computing.